src/app/CommandHandlerDispatcher.cpp

```diff
@@ -0,0 +1,128 @@
+#include "CommandHandlerDispatcher.h"
+#include "InteractionModelEngine.h"
+#include "messaging/ExchangeContext.h"
+
+namespace chip {
+namespace app {
+using Status = Protocols::InteractionModel::Status;
+
+CHIP_ERROR CommandHandlerDispatcher::OnMessageReceived(Messaging::ExchangeContext * apExchangeContext,
+                                                       const PayloadHeader & aPayloadHeader, System::PacketBufferHandle && aPayload)
+{
+    CHIP_ERROR err                           = CHIP_NO_ERROR;
+    bool sendStatusResponseWithInvalidAction = false;
+
+    if (mState == State::AwaitingResponse && aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::StatusResponse))
+    {
+        CHIP_ERROR statusError = CHIP_NO_ERROR;
+        err                    = StatusResponse::ProcessStatusResponse(std::move(aPayload), statusError);
+        VerifyOrExit(err == CHIP_NO_ERROR, sendStatusResponseWithInvalidAction = true);
+        statusError = err;
+        VerifyOrExit(err == CHIP_NO_ERROR, sendStatusResponseWithInvalidAction = true);
+
+        err = SendCommandResponse();
+        VerifyOrExit(err == CHIP_NO_ERROR, sendStatusResponseWithInvalidAction = true);
+
+        if (mState != State::AwaitingResponse)
+        {
+            // We are sending out the last message and no longer are expecting any responses. As a results we are
+            // fulfilling out responsibility to call close by calling ExitNow().
```

Comment:
```suggestion
            // We are sending out the last message and no longer are expecting any responses. As a result we are
            // fulfilling our responsibility to call close by calling ExitNow().
```
Please proofread your comments more carefully...